### PR TITLE
Add joker.test and More Eval Tests

### PR DESCRIPTION
**Pull request by jcburley**
This seems like a good point to pause and consider next steps for this work, which has some issues to sort out, mainly:

- How best to handle Clojure tests that don't currently work (or might never work) in Joker?
- What to do about `dosync` and `commute`?

My preference would be to keep `joker.test`, and especially the tests themselves, as close to their Clojure (and ClojureScript) counterparts as possible, easing comparison, enhancement, and so on.

But that raises both issues above. For `dosync` and `commute`, it seemed reasonable to preserve the places they're used, and just dummy them out -- in which case, those "dummy" versions should probably go into `joker.core` and perhaps be enhanced to participate in linting and such (plus `commute` could detect the type of argument(s) and handle them appropriately, rather than assuming the `atom` case).

For individual tests, at the `deftest` level (and higher, including helper functions), they were entirely removed; at finer-grained levels, lines were commented out.

I'm not sure there's a great alternative to commenting out tests-within-tests (other than removing those lines entirely).

At the `deftest` level, though, options include adding a `:disabled :some-reason` metadata to them (instead of simply removing them); or defining them via e.g. `deftest-disabled`. Either approach would require more editing to rename things like `Exception` (to `Error`) to ensure the tests are parsed successfully, even though they wouldn't get run.

Once an approach is decided upon, it should be documented (in a `README.md` inside `tests/eval/`, perhaps) and more tests can then be brought over consistent with that approach.

### CODE REVIEW

**Comment by jcburley** [comment]
But I think this PR is ready to merge and any discussion can lead to changes to make later on?

**Comment by jcburley** [comment]
Another issue I forgot to mention is what to do about the (commented-out) code that prints the stacktrace file/line info. I don't know (offhand) that Joker provides access to this sort of info or that we really need it (the remaining info has been enough for me to locate failing tests so far).

File: core/data/test.joke [hunk 1]
```diff
@@ -0,0 +1,781 @@
...
+   of *test-out*.
+
+   For additional event types, see the examples in the code.
+"}
+  joker.test
+  (:require [joker.template :as temp]
+            [joker.walk :as walk]))
+
+;; 2018-12-25 jcburley: Should this go into joker.core?
+(defn- dosync
```
[the remark this answers is not in the document]
Comment (jcburley) [reply core/data/test.joke]: Done.

**Comment by candid82** [comment]
Thanks!

**Pull request merged**